packages/@aws-cdk/aws-codedeploy/lib/deployment-group.ts

```diff
@@ -244,6 +291,62 @@ export class ServerDeploymentGroup extends ServerDeploymentGroupRef {
         break;
     }
   }
+
+  private ec2TagSet(tagSet?: InstanceTagSet):
+      cloudformation.DeploymentGroupResource.EC2TagSetProperty | undefined {
+    if (!tagSet || tagSet.instanceTagGroups.length === 0) {
+      return undefined;
+    }
+
+    return {
+      ec2TagSetList: tagSet.instanceTagGroups.map(tagGroup => {
+        return {
+          ec2TagGroup: this.tagGroup2TagsArray(tagGroup) as
+            cloudformation.DeploymentGroupResource.EC2TagFilterProperty[],
+        };
+      }),
+    };
+  }
+
+  private onPremiseTagSet(tagSet?: InstanceTagSet):
+      cloudformation.DeploymentGroupResource.OnPremisesTagSetProperty | undefined {
+    if (!tagSet || tagSet.instanceTagGroups.length === 0) {
+      return undefined;
+    }
+
+    return {
+      onPremisesTagSetList: tagSet.instanceTagGroups.map(tagGroup => {
+        return {
+          onPremisesTagGroup: this.tagGroup2TagsArray(tagGroup) as
+            cloudformation.DeploymentGroupResource.TagFilterProperty[],
+        };
+      }),
+    };
+  }
+
+  private tagGroup2TagsArray(tagGroup: InstanceTagGroup): any[] {
+    const tagsInGroup = [];
+    for (const tagKey in tagGroup) {
+      if (tagGroup.hasOwnProperty(tagKey)) {
+        const tagValues = tagGroup[tagKey];
+        if (tagValues.length > 0) {
+          for (const tagValue of tagValues) {
+            tagsInGroup.push({
+              key: tagKey,
+              value: tagValue,
+              type: 'KEY_AND_VALUE',
+            });
+          }
+        } else {
+          tagsInGroup.push({
+            key: tagKey,
+            type: 'KEY_AND_VALUE',
```

Comment:
should this be KEY_ONLY?

Reply:
And also, are we going to support VALUE_ONLY type tag?

Reply:
We absolutely should! Thanks for catching this.